cpp/arcticdb/storage/mongo/mongo_mock_client.hpp

```diff
@@ -0,0 +1,92 @@
+/* Copyright 2023 Man Group Operations Limited
+ *
+ * Use of this software is governed by the Business Source License 1.1 included in the file licenses/BSL.txt.
+ *
+ * As of the Change Date specified in that file, in accordance with the Business Source License, use of this software will be governed by the Apache License, version 2.0.
+ */
+
+#pragma once
+
+#include <arcticdb/storage/storage_utils.hpp>
+#include <arcticdb/storage/mongo/mongo_client_wrapper.hpp>
+#include <arcticdb/storage/storage_mock_client.hpp>
+#include <mongocxx/exception/operation_exception.hpp>
+
+namespace arcticdb::storage::mongo {
+
+struct MongoFailure {
```

Comment:
An alias might help to make the code "self-documenting" here,

```
using NoAckFailure = std::monostate;
```

Reply:
Fixed in [3283f32](https://github.com/man-group/ArcticDB/pull/1395/commits/3283f32c0a323b10940abd25d0b876ca8b57984e)